### DISCUSSION TITLE
Migrating bookmarks between VSCode instances

### DISCUSSION THREAD

**Discussion by ldelossa** [alefragnani/vscode-bookmarks#592] [Q&A] [answered]
Hello, 

I typically use two different laptops. Each laptop have repositories in the same absolute path. 

Is it possible to sync bookmarks between VSCode instances? I imagine this can be done even with the absolute paths do not match? As long as the relative root dir name is the same?

**Comment by alefragnani** [comment, marked as answer]
Hi @ldelossa ,

There is no way to _sync_ bookmarks between different machines, as the native Settings Sync feature does not support syncing extension’s data.

But, as an alternative, you could take a look at the `bookmarks.saveBookmarksInProject` setting (https://github.com/alefragnani/vscode-bookmarks#available-settings), which will make the bookmarks to be stored within the project/workspace you have opened. If the repositories you are referring to are Git repos, for instance, you just have to add the `.vscode/bookmarks.json` file to the repo `push`/`pull` to update the other machines.

Hope this helps

**Reply by ldelossa** [reply]
oh interesting! is it possible to save them to my dotfiles repository?

**Reply by alefragnani** [reply]
No, it won’t be possible. The extension uses a hard coded location, the `.vscode` folder, that VS Code uses to store some files. I just borrowed it 😄 .

If you think your dotfiles repository would be an alternative, I understand you would expect to have _all your bookmarks stored on an unique location_. Because the extension uses a _per session approach_, the bookmarks are stored individually, for every folder. But, there is a feature request asking for _global bookmarks_ (https://github.com/alefragnani/vscode-bookmarks/issues/287). This could be a step to some _external location_, like dotfiles.

**Reply by ldelossa** [reply]
Hmm, in the least, can i just copy and paste some files from both computer? 

I work in a lot of public repos which storing the bm files in the .vscode file and pushing it to git is not an option.

**Reply by alefragnani** [reply]
Other than `.vscode/bookmarks.json` no, there isn’t. Because the other location is the VS Code internals, which is a SQLite database on each workspace cache.

If/when the _global bookmarks_ is available, it could be expanded to allow an external storage.

**Reply by ldelossa** [reply]
Okay, thanks a lot :)
